### PR TITLE
Updating the faq

### PR DESCRIPTION
**Pull request by sfshaza2**
Staged: https://sz-flutter.firebaseapp.com/docs/resources/faq

I started this PR to specifically address https://github.com/flutter/website/issues/4039.

However, as I scanned through it, it is in serious need of updating. I updated some of it, but did not tackle everything.

@jonahwilliams, please look at the "What devices does Flutter run on?" section, which is the reason that the doc bug was filed.

@timsneath, as I said, I've tackled *some* of the rest of this doc, but maybe a PM could do a more thorough pass on it? In a new PR, perhaps?  :)

### CODE REVIEW

File: src/docs/resources/faq.md [hunk 1]
```diff
@@ -416,59 +415,62 @@ Hot reload works by injecting updated source code files
...
 
-Mobile hardware: iOS devices (iPhone 4S or newer) and ARM Android devices.
+* Mobile hardware: iOS devices (iPhone 4S or newer)
+  and ARM Android devices.
 
-Note Flutter currently does not support building for x86 Android
-([issue #9253][]) directly, however apps built for ARMv7 or
-ARM64 run fine (via ARM emulation) on many x86 Android devices.
+* Flutter supports building ahead-of-time (AOT) compiled libraries
+  for `x86_64`, `armeabi-v7a`, and `arm64-v8a`.
```
Comment (jonahwilliams) [inline src/docs/resources/faq.md:440]: The note about ARM emulation for x86 devices is still important

File: src/docs/resources/faq.md [hunk 2]
```diff
@@ -416,59 +415,65 @@ Hot reload works by injecting updated source code files
...
-We test on a variety of low-end to high-end phones
-but we don't yet have an official device compatibility guarantee.
+* We test on a variety of low-end to high-end phones
+  but we don't yet have an official device compatibility guarantee.
 
-We believe Flutter works well on tablets. We do not currently implement
-all of the tablet-specific adaptations recommended by Material Design,
-though we are planning further investment in this area.
+* We believe Flutter works well on tablets. We do not
+  currently implement all of the tablet-specific adaptations
```
Comment (timsneath) [inline src/docs/resources/faq.md:452]: I'm not sure if this is still true. We should check with @willlarche whether this call-out is necessary. I suspect it dates back from a much earlier time in Flutter's Material development.

File: src/docs/resources/faq.md [hunk 3]
```diff
@@ -416,59 +415,65 @@ Hot reload works by injecting updated source code files
...
 
-We test on a variety of low-end to high-end phones
-but we don't yet have an official device compatibility guarantee.
+* We test on a variety of low-end to high-end phones
+  but we don't yet have an official device compatibility guarantee.
 
-We believe Flutter works well on tablets. We do not currently implement
-all of the tablet-specific adaptations recommended by Material Design,
-though we are planning further investment in this area.
+* We believe Flutter works well on tablets. We do not
```
Comment (timsneath) [inline src/docs/resources/faq.md:451]: We can be stronger than "We _believe_ Flutter works well on tablets". There's nothing fundamentally different about iPadOS or Android on tablets in terms of its compilation / execution, and we test on those devices just like we test on phones.

File: src/docs/resources/faq.md [hunk 4]
```diff
@@ -570,19 +575,20 @@ Learn more about a11y in the
...
-are able to run in parallel (usually implemented as separate threads). Isolates
-communicate by sending and receiving asynchronous messages. Flutter does not
-currently have a shared-memory parallelism solution, although we are evaluating
-solutions for this.
+Flutter supports isolates. Isolates are separate heaps in
+Flutter's VM, and they are able to run in parallel
+(usually implemented as separate threads). Isolates
+communicate by sending and receiving asynchronous messages.
+Flutter does not currently have a shared-memory parallelism
+solution, although we are evaluating solutions for this.
```
Comment (timsneath) [inline src/docs/resources/faq.md:583]: Let's remove this last sentence. It seems unnecessary to make a promise here.

File: src/docs/resources/faq.md (hunk 2 above)
Comment (willlarche) [reply src/docs/resources/faq.md:452]: I would say it's not necessary anymore. Happy to take community feedback on features they'd like tho!

**Pull request merged**